Tests/FunctionalTests/TestDiscoveryTests.swift

```diff
@@ -12,159 +12,185 @@
 
 import Basics
 import PackageModel
+import SPMBuildCore
 import _InternalTestSupport
-import XCTest
+import Testing
+import struct Foundation.UUID
+import class Foundation.ProcessInfo
 
-final class TestDiscoveryTests: XCTestCase {
-    func testBuild() async throws {
-        try await fixture(name: "Miscellaneous/TestDiscovery/Simple") { fixturePath in
-            let (stdout, _) = try await executeSwiftBuild(fixturePath)
-            // in "swift build" build output goes to stdout
-            XCTAssertMatch(stdout, .contains("Build complete!"))
+@Suite
+struct TestDiscoveryTests {
+    static var buildSystems: [BuildSystemProvider.Kind] = [BuildSystemProvider.Kind.native, .swiftbuild]
+
+    @Test(arguments: buildSystems)
+    func build(_ buildSystem: BuildSystemProvider.Kind) async throws {
+        try await withKnownIssue("Windows builds encounter long path handling issues", isIntermittent: true) {
+            try await fixture(name: "Miscellaneous/TestDiscovery/Simple") { fixturePath in
+                let (stdout, _) = try await executeSwiftBuild(fixturePath, buildSystem: buildSystem)
+                // in "swift build" build output goes to stdout
+                #expect(stdout.contains("Build complete!"))
+            }
+        } when: {
+            buildSystem == .swiftbuild && ProcessInfo.hostOperatingSystem == .windows
         }
     }
 
-    func testDiscovery() async throws {
-        try await fixture(name: "Miscellaneous/TestDiscovery/Simple") { fixturePath in
-            let (stdout, stderr) = try await executeSwiftTest(fixturePath)
-            // in "swift test" build output goes to stderr
-            XCTAssertMatch(stderr, .contains("Build complete!"))
-            // in "swift test" test output goes to stdout
-            XCTAssertMatch(stdout, .contains("Executed 3 tests"))
+    @Test(arguments: buildSystems)
+    func discovery(_ buildSystem: BuildSystemProvider.Kind) async throws {
+        try await withKnownIssue("Windows builds encounter long path handling issues", isIntermittent: true) {
+            try await fixture(name: "Miscellaneous/TestDiscovery/Simple") { fixturePath in
+                let (stdout, stderr) = try await executeSwiftTest(fixturePath, extraArgs: ["-vv"], buildSystem: buildSystem)
+                // in "swift test" build output goes to stderr
+                #expect(stderr.contains("Build complete!"))
+                // in "swift test" test output goes to stdout
+                #expect(stdout.contains("Executed 3 tests"))
+            }
+        } when: {
+            buildSystem == .swiftbuild && ProcessInfo.hostOperatingSystem == .windows
         }
     }
 
-    func testNonStandardName() async throws {
+    @Test(.bug("https://github.com/swiftlang/swift-build/issues/13"), arguments: [BuildSystemProvider.Kind.native])
```

Comment:
question: out of curiosity, why aren't we executing this test against the SwiftBuild build system?